tests/mock_crust/mod.rs

```diff
@@ -213,77 +208,144 @@ fn multiple_clients_per_proxy() {
-    let leaky_rate = 2 * MAX_IMMUTABLE_DATA_SIZE_IN_BYTES;
-    let per_client_cap = CAPACITY / MAX_CLIENTS_PER_PROXY as u64;
-    for i in 0..10 {
-        trace!("iteration {:?}", i);
-        let mut clients_sent = HashMap::new();
-        for client in &mut clients {
-            if rng.gen_weighted_bool(2) {
-                let msg_id = MessageId::new();
-                unwrap!(client.inner.get_idata(dst, data_id, msg_id));
-                let _ = clients_sent.insert(msg_id, client.ip());
-            }
+
+    let mut clients_sent = HashMap::new();
+    for client in clients.iter_mut() {
+        if is_always_send || rng.gen_weighted_bool(2) {
+            let msg_id = MessageId::new();
+            unwrap!(client.inner.get_idata(dst, data_id, msg_id));
+            let _ = clients_sent.insert(msg_id, client.ip());
         }
-        trace!("clients_sent: {:?}", clients_sent);
-        let _ = poll_all(&mut nodes, &mut clients);
-
-        let mut request_received: HashMap<MessageId, usize> = HashMap::new();
-        for node in nodes.iter_mut().filter(|n| n.is_recipient(&dst)) {
-            while let Ok(event) = node.try_next_ev() {
-                if let Event::Request {
-                           request: Request::GetIData { msg_id: req_message_id, .. }, ..
-                       } = event {
-                    let entry = request_received
-                        .entry(req_message_id)
-                        .or_insert_with(|| 0);
-                    *entry += 1;
-                }
+    }
+    trace!("clients_sent: {:?}", clients_sent);
+    let _ = poll_all(nodes, clients);
+
+    let mut request_received: HashMap<MessageId, usize> = HashMap::new();
+    for node in nodes.iter_mut().filter(|n| n.is_recipient(&dst)) {
+        while let Ok(event) = node.try_next_ev() {
+            if let Event::Request {
+                       request: Request::GetIData { msg_id: req_message_id, .. }, ..
+                   } = event {
+                let entry = request_received
+                    .entry(req_message_id)
+                    .or_insert_with(|| 0);
+                *entry += 1;
             }
         }
-        trace!("request_received: {:?}", request_received);
+    }
+    trace!("request_received: {:?}", request_received);
+
+    for (msg_id, count) in &request_received {
+        assert_eq!(*count, MIN_SECTION_SIZE);
+        let _ = unwrap!(clients_sent.remove(msg_id));
+        *total_usage += MAX_IMMUTABLE_DATA_SIZE_IN_BYTES;
+    }
+    assert!(*total_usage <= CAPACITY);
+
+    clients_sent
+}
 
-        for (msg_id, count) in &request_received {
-            assert_eq!(*count, min_section_size);
-            let _ = unwrap!(clients_sent.remove(msg_id));
-            total_usage += MAX_IMMUTABLE_DATA_SIZE_IN_BYTES;
+// Verifies the usage in rate_limiter. Also advances the fake clock.
+fn rate_limiter_verify(clients_sent: &HashMap<MessageId, IpAddr>,
```

Comment:
I would rename `clients_sent` here and the few instances below to indicate it contains only rejected messages (e.g. `rejected_msgs`?)